types/ember/index.d.ts

```diff
@@ -3311,7 +3311,22 @@ declare module 'ember' {
         /**
          * Returns a consistent type for the passed object.
          */
-        function typeOf(item?: any): string;
+        function typeOf(item: string | String): 'string';
```

Comment:
I'd suggest doing something like this:
```ts
type KeysOfType<Base, Condition> = keyof Pick<Base, {
    [Key in keyof Base]: Base[Key] extends Condition ? Key : never
}[keyof Base]>;

type TypeLookup = {
    regexp: RegExp;
    boolean: boolean;
    date: Date;
};

function typeOf<T>(value: T): KeysOfType<TypeLookup, T> { return "boolean" as any };

typeOf(true); // "boolean"
typeOf(null as (RegExp | Date)); // "regexp" | "date" 
```
This lets union types of values flow through into union types of strings

Reply:
Great suggestion. Thanks!